### PR TITLE
Build caddy with plugins via multi-stage

### LINKED ISSUE #120: Build of container failed
Caddy has updated their website and doesn't use the same url. 
+ the locale plugin is not (yet) on the new website (https://github.com/simia-tech/caddy-locale/issues/4)
+ seems to be the same for https://github.com/pedronasser/caddy-search 

https://drone.gitea.io/go-gitea/docs/323

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/45831088-build-of-container-failed?utm_campaign=plugin&utm_content=tracker%2F47955789&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F47955789&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sapk**
same as go-gitea/website#43 +  go-gitea/website#45

This will need a new drone sig (cc @lunny @appleboy and anyone who can sign ^^)

Fix #120

### CODE REVIEW

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

**Comment by lunny** [comment]
Have we changed drone to 0.6 now?

**Comment by appleboy** [comment]
@lunny No. still needs to update drone sig file.

**Comment by bkcsoft** [comment]
Does this work with the docker-version we have on demo? (v1.13.1 API 1.26)

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+RUN apk add --no-cache git 
+RUN go get -d -v github.com/mholt/caddy/caddy github.com/pedronasser/caddy-search github.com/simia-tech/caddy-locale
+WORKDIR /go/src/github.com/mholt/caddy/caddy
+
+RUN sed -i '/This is where other plugins get plugged in (imported)/a _ "github.com/pedronasser/caddy-search"' caddymain/run.go \
+ && sed -i '/This is where other plugins get plugged in (imported)/a _ "github.com/simia-tech/caddy-locale"' caddymain/run.go \
+ && go install -v . \
+ && /go/bin/caddy -version
+
 FROM alpine:edge
```
Comment (bkcsoft) [inline Dockerfile:13]: Does docker v1.13.1 have support for multi-image?

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (bkcsoft) [reply Dockerfile:13]: (anyone can dismiss this review if this does work...)

Comment (sapk) [reply Dockerfile:13]: We only need Docker > 17.05 for building but the resulting image is compatible with any old version (only the last image is kept). This function permit to use a builder pattern in one Dockerfile (with possible multi-builder).

Comment (sapk) [reply Dockerfile:13]: For confirmation, this is already in production on https://gitea.io with the same solution.

Comment (bkcsoft) [reply Dockerfile:13]: Well, it seems to bulid do :)

**Review by bkcsoft — APPROVED** [review]

**Pull request merged**